packages/PostRector/Collector/NodesToAddCollector.php

```diff
@@ -132,14 +142,22 @@ private function resolveNearestExpressionPosition(Node $node): string
             return spl_object_hash($node);
         }
 
+        $currentStmt = $node->getAttribute(AttributeKey::CURRENT_STATEMENT);
+        if ($currentStmt instanceof Stmt) {
+            return spl_object_hash($currentStmt);
+        }
+
         $parent = $node->getAttribute(AttributeKey::PARENT_NODE);
         if ($parent instanceof Return_) {
             return spl_object_hash($parent);
         }
 
         $foundNode = $this->betterNodeFinder->findParentTypes($node, [Expression::class, Stmt::class]);
-        if ($foundNode === null) {
-            $foundNode = $node;
```

Comment:
Not sure why I put this here (probably not knowing difference between Stmt and Expr in that time), but it was asking for a bug :thinking: